### PR TITLE
Add seventh problem

### PR DESCRIPTION
**Pull request by helenKli**

### CODE REVIEW

File: 7. Reverse Integer/code.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <iostream>
+#include <climits>
+
+int reverse(int x) {
+    int rev = 0;
+    while (x != 0) {
+        int y = x % 10;
+        x /= 10;
+        if (rev > INT_MAX/10 || (rev == INT_MAX / 10 && y > 7)) return 0;
```
Comment (KlicOgogo) [inline 7. Reverse Integer/code.cpp]: Add spaces for INT_MAX/10

File: 7. Reverse Integer/code.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    int result = reverse(x);
+    if (result == correctAnswer) {
+        std::cout << "Test passed" << std::endl;
+    } else {
+        std::cout << "Test not passed" << std::endl;
+    }
+}
+
+int main() {
+    test(1234, 4321);
```
Comment (KlicOgogo) [inline 7. Reverse Integer/code.cpp:27]: Add test with corner cases like 1.123.456.789. Also with negative numbers

File: 7. Reverse Integer/code.cpp (hunk 1 above)
Comment (KlicOgogo) [reply 7. Reverse Integer/code.cpp]: Update algorithm with using `long long int` type

**Review by KlicOgogo — CHANGES_REQUESTED** [review]

File: 13. Roman to Integer/code.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    return sum;
+}
+
+void test(std::string str, int correctAnswer) {
+    int result = romanToInt(str);
+    if (result == correctAnswer) {
+        std::cout << "Test passed" << std::endl;
+    } else {
+        std::cout << "Test not passe" << std::endl;
```
Comment (KlicOgogo) [inline 13. Roman to Integer/code.cpp]: Test not passed

File: 13. Roman to Integer/code.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                    i++;
+                    sum += 4;
+                }
+                else sum += 1;
+        }
+    }
+    return sum;
+}
+
+void test(std::string str, int correctAnswer) {
```
Comment (KlicOgogo) [inline 13. Roman to Integer/code.cpp]: Use const std::string& instead of std::string. 
See https://stackoverflow.com/questions/2582797/why-pass-by-const-reference-instead-of-by-value

File: 13. Roman to Integer/code.cpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            sum += 50;
+        }
+        if (s[i] == 'X') {
+            if (s[i+1] == 'C') {
+                i++;
+                sum += 90;
+            }   else if (s[i+1] == 'L') {
+                    i++;
+                    sum += 40;
+                }   else sum += 10;
```
Comment (KlicOgogo) [inline 13. Roman to Integer/code.cpp]: Use else {
} even for 1-line expressions. Also check code formatting (something is wrong with braces.

File: 14. Longest Common Prefix/code.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
+//https://leetcode.com/problems/longest-common-prefix/
+
+#include <iostream>
+#include <vector>
+
+std::string longestCommonPrefix(std::vector<std::string>& strs) {
```
Comment (KlicOgogo) [inline 14. Longest Common Prefix/code.cpp]: const std::vector<..>& strs

File: 14. Longest Common Prefix/code.cpp [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        }
+        if (n == 0) {
+            return "";
+        }
+    }
+    res.erase(n, res.length() - n);
+    return res;
+}
+
+void test(std::vector<std::string> str, std::string correctAnswer) {
```
Comment (KlicOgogo) [inline 14. Longest Common Prefix/code.cpp]: Also use const references

File: 20. Valid Parentheses/code.cpp [hunk 8]
```diff
@@ -0,0 +1,39 @@
+//https://leetcode.com/problems/valid-parentheses/
+
+#include <iostream>
+#include <stack>
+
+bool isValid(std::string s) {
+    std::stack <char> steck;
```
Comment (KlicOgogo) [inline 20. Valid Parentheses/code.cpp:7]: std::stack<char> st;

File: 20. Valid Parentheses/code.cpp [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    for (int i = 0; i < s.length(); i++) {
+        if (s[i] == '(' || s[i] == '{' || s[i] == '[') {
+            steck.push(s[i]);
+        }
+        if (s[i] == ')' || s[i] == '}' || s[i] == ']') {
+            if (steck.empty()) {
+                return false;
+            }
+            switch (s[i]) {
+                case ')': if(steck.top() == '('){steck.pop();} else return false; break;
```
Comment (KlicOgogo) [inline 20. Valid Parentheses/code.cpp:17]: Using several lines will be more readable

File: 21. Merge Two Sorted Lists/.vscode/settings.json [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (KlicOgogo) [inline 21. Merge Two Sorted Lists/.vscode/settings.json]: This files should be removed
Add .vscode/ to .gitignore

File: 21. Merge Two Sorted Lists/code.cpp [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    ListNode *next;
+    ListNode() : val(0), next(nullptr) {}
+    ListNode(int x) : val(x), next(nullptr) {}
+    ListNode(int x, ListNode *next) : val(x), next(next) {}
+ };
+
+ ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
+    ListNode* Head = nullptr;
+    ListNode* step = nullptr;
+    if(l1 == nullptr){
```
Comment (KlicOgogo) [inline 21. Merge Two Sorted Lists/code.cpp]: Add spaces after )

File: 21. Merge Two Sorted Lists/code.cpp [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+        return l2;
+    }
+    if(l2 == nullptr){
+        return l1;
+    }
+    while(l1 != nullptr || l2 != nullptr){
+        if(step == nullptr){
```
Comment (KlicOgogo) [inline 21. Merge Two Sorted Lists/code.cpp]: Add spaces before (

File: 21. Merge Two Sorted Lists/code.cpp [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+struct ListNode {
+    int val;
+    ListNode *next;
+    ListNode() : val(0), next(nullptr) {}
+    ListNode(int x) : val(x), next(nullptr) {}
+    ListNode(int x, ListNode *next) : val(x), next(next) {}
+ };
+
+ ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
+    ListNode* Head = nullptr;
```
Comment (KlicOgogo) [inline 21. Merge Two Sorted Lists/code.cpp]: ListNode* head

File: 21. Merge Two Sorted Lists/code.cpp [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+        } else {
+            break;
+        }
+    }
+    if (result != nullptr || correctAnswer != nullptr) {
+        std::cout << "Test not passed" << std::endl;
+    } else {
+        std::cout << "Test passed" << std::endl;
+    }
+    /*if (result == correctAnswer){
```
Comment (KlicOgogo) [inline 21. Merge Two Sorted Lists/code.cpp]: Remove this commented code

File: 9. Palindrome Number/code.cpp [hunk 15]
```diff
@@ -0,0 +1,34 @@
+//https://leetcode.com/problems/palindrome-number/
+
+#include <iostream>
+
+bool isPalindrome(int x) {
+    if (x < 0 || (x % 10 == 0 && x != 0)) {
+        return 0;
```
Comment (KlicOgogo) [inline 9. Palindrome Number/code.cpp]: return true or false, not 1 or 0

File: 14. Longest Common Prefix/code.cpp [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+        }
+        if (n == 0) {
+            return "";
+        }
+    }
+    res.erase(n, res.length() - n);
+    return res;
+}
+
+void test(const std::vector<std::string> str, const std::string correctAnswer) {
```
Comment (KlicOgogo) [inline 14. Longest Common Prefix/code.cpp]: Add reference

Comment (KlicOgogo) [reply 14. Longest Common Prefix/code.cpp]: Like const std:string&

File: 9. Palindrome Number/.vscode/settings.json [hunk 17]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (KlicOgogo) [inline 9. Palindrome Number/.vscode/settings.json]: Should be under gitignore

File: 14. Longest Common Prefix/code.cpp [hunk 18]
```diff
@@ -0,0 +1,43 @@
+//https://leetcode.com/problems/longest-common-prefix/
+
+#include <iostream>
+#include <vector>
+
+std::string longestCommonPrefix(const std::vector<std::string>& strs) {
```
Comment (KlicOgogo) [inline 14. Longest Common Prefix/code.cpp:6]: strs is passed correctly, but check all functions with vectors and strings again

**Review by KlicOgogo — CHANGES_REQUESTED** [review]

**Comment by KlicOgogo** [comment]
Use `return 0;` in main
Add empty line at the end of each file
Check function parameters

File: 26. Remove Duplicates from Sorted Array/code.cpp [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+        if (nums[i] != k) {
+            k = nums[i];
+            nums[count] = nums[i];
+            count++;
+        }
+    }
+    return count;
+}
+
+void test(std::vector<int>& nums, const int& correctAnswer) {
```
Comment (KlicOgogo) [inline 26. Remove Duplicates from Sorted Array/code.cpp]: int correctAnswer

File: 67. Add Binary/code.cpp [hunk 20]
```diff
@@ -0,0 +1,79 @@
+//https://leetcode.com/problems/add-binary/
+
+#include <iostream>
+#include <string>
+
+std::string addBinary(std::string& a, std::string& b) {
```
Comment (KlicOgogo) [inline 67. Add Binary/code.cpp]: I think they should be const, also in test

**Review by KlicOgogo — APPROVED** [review]
